lxd/daemon.go

```diff
@@ -322,6 +322,16 @@ func (d *Daemon) Authenticate(r *http.Request) (bool, string, string, error) {
 		}
 	}
 
+	secret := r.Header.Get("X-LXD-secret")
+	fingerprint := r.Header.Get("X-LXD-fingerprint")
+
+	if secret != "" && fingerprint != "" {
+		valid := imageValidSecret(fingerprint, secret)
+		if valid {
+			return true, "", "secret", nil
+		}
+	}
```

Comment:
This feels out of place here.

So far the locations where secrets were used were using a syntax like `?secret=STRING`.

We already have support for unauthenticated access to some of the operation endpoints using this syntax so I think we should keep on doing that.

Reply:
This code allows the source to call `CreateImage` on the target. If we were to use `?secret=blah` here, we'd need to allow untrusted clients to post to `/1.0/images`. Is that a good idea?

Reply:
Hmm, right, so using a field in the struct is problematic as they could send us a lot of data and cause a DoS.

Using the headers seem fine then. I wouldn't do the check in `daemon.go` though, we should move the access check in the imagesPost function and have it be very struct, such that if we're not dealing with a trusted client, we validate the headers, if the headers aren't there or aren't valid we immediately reject the request. We should never read the body prior to authentication.